itest/lnd_route_blinding.go

```diff
@@ -0,0 +1,298 @@
+	aliceBobChan := ht.GetChannelByChanPoint(alice, chanPointAliceBob)
+	bobCarolChan := ht.GetChannelByChanPoint(bob, chanPointBobCarol)
+
+	// Sanity check that bob's fee is as expected.
+	chanInfoReq := &lnrpc.ChanInfoRequest{
+		ChanId: bobCarolChan.ChanId,
+	}
+
+	bobCarolInfo, err := bob.RPC.LN.GetChanInfo(ctxb, chanInfoReq)
+	require.NoError(ht, err)
+
+	// Our test relies on knowing the fee rate for bob - carol to set the
+	// fees we expect for our route. Perform a quick sanity check that our
+	// policy is as expected.
+	var policy *lnrpc.RoutingPolicy
+	if bobCarolInfo.Node1Pub == bob.PubKeyStr {
+		policy = bobCarolInfo.Node1Policy
+	} else {
+		policy = bobCarolInfo.Node2Policy
+	}
+	require.Equal(ht, bobCarolBase, uint64(policy.FeeBaseMsat), "base fee")
+	require.EqualValues(ht, 0, policy.FeeRateMilliMsat, "fee rate")
+
+	// We'll also need the current block height to calculate our locktimes.
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+
+	info, err := alice.RPC.LN.GetInfo(ctxt, &lnrpc.GetInfoRequest{})
+	require.NoError(ht, err)
+	cancel()
+
+	// Since we created channels with default parameters, we can assume
+	// that all of our channels have the default cltv delta.
+	bobCarolDelta := uint32(chainreg.DefaultBitcoinTimeLockDelta)
+
+	// Create arbitrary pubkeys for use in our blinded route. They're not
+	// actually used functionally in this test, so we can just make them up.
+	var (
+		_, blindingPoint = btcec.PrivKeyFromBytes([]byte{1})
+		_, carolBlinded  = btcec.PrivKeyFromBytes([]byte{2})
+		_, blindedHop1   = btcec.PrivKeyFromBytes([]byte{3})
+		_, blindedHop2   = btcec.PrivKeyFromBytes([]byte{4})
+
+		encryptedDataCarol = []byte{1, 2, 3}
+		encryptedData1     = []byte{4, 5, 6}
+		encryptedData2     = []byte{7, 8, 9}
+
+		blindingBytes     = blindingPoint.SerializeCompressed()
+		carolBlindedBytes = carolBlinded.SerializeCompressed()
+		blinded1Bytes     = blindedHop1.SerializeCompressed()
+		blinded2Bytes     = blindedHop2.SerializeCompressed()
+	)
+
+	// Now we create a blinded route which uses carol as an introduction
+	// node followed by two dummy hops (the arbitrary pubkeys in our
+	// blinded route above:
+	// Carol --- B1 --- B2
+	route := &lnrpc.BlindedRoute{
+		IntroductionNode: carol.PubKey[:],
+		BlindingPoint:    blindingBytes,
+		BlindedHops: []*lnrpc.BlindedHop{
+			{
+				// The first hop in the blinded route is
+				// expected to be the introduction node.
+				BlindedNode:   carolBlindedBytes,
+				EncryptedData: encryptedDataCarol,
+			},
+			{
+				BlindedNode:   blinded1Bytes,
+				EncryptedData: encryptedData1,
+			},
+			{
+				BlindedNode:   blinded2Bytes,
+				EncryptedData: encryptedData2,
+			},
+		},
+	}
+
+	// Create a blinded payment that has aggregate cltv and fee params
+	// for our route.
+	var (
+		aggregateBaseFee   uint64 = 1500
+		aggregateCltvDelta uint32 = 125
+		cltvLimit                 = info.BlockHeight + 500
```

Comment:
I think cltvLimit is a delta so no need to add blockheight, at least that's what `maxTotalTimelock` (which we take when no cltvLimit is set) is, which is default 2016 blocks